omnigibson/prims/rigid_body_prim.py

```diff
@@ -0,0 +1,271 @@
+import math
+from typing import Literal
+
+import torch as th
+
+import omnigibson as og
+import omnigibson.lazy as lazy
+from omnigibson.utils.usd_utils import PoseAPI
+
+from .rigid_prim import RigidPrim
+
+
+class RigidBodyPrim(RigidPrim):
+    """
+    Provides high level functions to deal with a dynamic rigid body prim and its attributes/properties.
+    This class is used for rigid bodies that are not kinematic-only, meaning they are subject to physics simulation
+    dynamics like gravity, forces, and collisions.
+
+    Args:
+        relative_prim_path (str): Scene-local prim path of the Prim to encapsulate or create.
+        name (str): Name for the object. Names need to be unique per scene.
+        load_config (None or dict): If specified, should contain keyword-mapped values that are relevant for
+            loading this prim at runtime.
+    """
+
+    def __init__(
```

Comment:
i assume the bulk of this is copied vertabim from the original RigidPrim class lol